homeassistant/components/asuswrt/config_flow.py

```diff
@@ -0,0 +1,175 @@
+"""Config flow to configure the AsusWrt integration."""
+import logging
+import os
+import socket
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MODE,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PROTOCOL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers import config_validation as cv
+
+# pylint:disable=unused-import
+from .const import (
+    CONF_DNSMASQ,
+    CONF_INTERFACE,
+    CONF_REQUIRE_IP,
+    CONF_SSH_KEY,
+    DEFAULT_DNSMASQ,
+    DEFAULT_INTERFACE,
+    DEFAULT_SSH_PORT,
+    DOMAIN,
+)
+from .router import get_api
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_file(value) -> bool:
+    """Validate that the value is an existing file."""
+    file_in = os.path.expanduser(str(value))
+
+    if not os.path.isfile(file_in):
+        return False
+    if not os.access(file_in, os.R_OK):
+        return False
+    return True
+
+
+def _get_ip(host):
+    """Get the ip address from the host name."""
+    if host is None:
+        return None
+    try:
+        return socket.gethostbyname(host)
+    except socket.gaierror:
+        return None
+
+
+class AsusWrtFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize AsusWrt config flow."""
+        self._host = None
+        self._name = None
+
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
```

Comment:
Its a lot of parameters for one step, is there no logical way to break this up into multiple steps?

Reply:
The only logic that I can see is that selecting Telnet as protocol SSH key does not make sense. Does make sense separate this?

Reply:
Sounds like unnecessary layer for one option